context/app/api/vitessce_confs/assay_confs.py

```diff
@@ -340,8 +379,15 @@ def get_view_config_class_for_data_types(entity, nexus_token):
             entity=entity, nexus_token=nexus_token)
     if "rna" in hints:
         # This is the zarr-backed anndata pipeline.
-        if('anndata-to-ui.cwl' in dag_names):
-            return RNASeqAnnDataZarrViewConfBuilder(entity=entity, nexus_token=nexus_token)
+        if "anndata-to-ui.cwl" in dag_names:
+            if "salmon_rnaseq_slideseq" in data_types:
+                return SpatialRNASeqAnnDataZarrViewConfBuilder(
+                    entity=entity,
+                    nexus_token=nexus_token
+                )
+            return RNASeqAnnDataZarrViewConfBuilder(
+                entity=entity, nexus_token=nexus_token
+            )
```

Comment:
Could you refactor this whole function so that it just returns the _class_, and leaves it to the caller to construct the _instance_? I think all the constructors except for `NullViewConfBuilder` have the same signature, so it shouldn't be a a huge pain. So on just these lines the change looks like:

```suggestion
        if "anndata-to-ui.cwl" in dag_names:
            if "salmon_rnaseq_slideseq" in data_types:
                return SpatialRNASeqAnnDataZarrViewConfBuilder
            return RNASeqAnnDataZarrViewConfBuilder
```
The if-then tree is getting unwieldy, so cutting the number of lines in half is a win for readability.

At the caller, the change would be something like be:

```diff
-vc = get_view_config_class_for_data_types(
-    entity=entity, nexus_token=self.nexus_token
-)
+Builder = get_view_config_builder(entity)
+vc = Builder(entity, self.nexus_token)
```

This is the sort of thing [Python 3.10's pattern matching](https://www.python.org/dev/peps/pep-0636/) might be applied to, but I don't think it's worth getting into that here.

Reply:
Yes!